theme/_router.ts

```diff
@@ -0,0 +1,19 @@
+import { match, route } from '@/theme-kit';
+import { assertServerEnv } from '@/utils';
+
+export default function router(path: string | string[]) {
+    assertServerEnv('router');
+
+    return match(path, [
+        route('/', () => import('./homepage')),
+        route('/[locale]', () => import('./homepage')),
+        route('/[slug]', () => import('./story')),
+        route('/s/[uuid]', () => import('./story')),
+        route('/media', () => import('./gallery')),
+        route('/[locale]/media', () => import('./gallery')),
+        route('/media/album/[uuid]', () => import('./gallery-album')),
+        route('/[locale]/media/album/[uuid]', () => import('./gallery-album')),
+        route('/search', () => import('./search')),
+        route('/[locale]/search', () => import('./search')),
```

Comment:
It feels a bit weird to have two route definitions.
I know that Next doesn't support a required path parameter after an optional one, but can we do this for static routes like this one?
```suggestion
        route('/[...locale]/search', () => import('./search')),
```

Reply:
Yeah, this is something I definitely wanted to improve here. Because this is actually custom syntax that only resembles Next.js routing conventions, this can be modified to allow optional path variables. Like so:

```
/[[locale]]/search
```

... and the router will just understand it. 

I didn't address it here, as it wasn't required for the minimal PoC I've been aiming for.